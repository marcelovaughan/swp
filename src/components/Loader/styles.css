#spinner {
    position: fixed;
    z-index: 102;
    top: 50%;
    left: 0;
    right: 0;
    margin-top: -40px;
  }
  
  #spinner .loader, .module-spinner .loader {
    display: block;
    position: relative;
    width: 40px;
    height: 40px;
    margin: 0 auto;
  }
  
  .loading-animation, #spinner .loader, .module-spinner .loader {
    background-position: center;
    background-image: url(https://static-mh.content.disney.io/starwars/assets/shared/loader_empire-985b0940808a.png);
    background-size: 100%;
    background-repeat: no-repeat;
  }
  
  .loading-animation:before, #spinner .loader:before, .module-spinner .loader:before {
    height: 52px;
    width: 52px;
    margin: -26px 0 0 -26px;
    animation: clockwise 5s infinite linear;
  }
  
  .loading-animation:before, #spinner .loader:before, .module-spinner .loader:before, .loading-animation:after, #spinner .loader:after, .module-spinner .loader:after {
    content: "";
    display: block;
    position: absolute;
    top: 50%;
    left: 50%;
    border-radius: 50%;
    border: 3px solid #fff;
    border-top-color: transparent;
    box-sizing: border-box;
  }
  
  .loading-animation:after, #spinner .loader:after, .module-spinner .loader:after {
    height: 66px;
    width: 66px;
    margin: -33px 0 0 -33px;
    border-width: 4px;
    border-left-color: transparent;
    animation: counter-clockwise 2.25s infinite linear;
  }
  
  .loading-animation:before, #spinner .loader:before, .module-spinner .loader:before, .loading-animation:after, #spinner .loader:after, .module-spinner .loader:after {
    content: "";
    display: block;
    position: absolute;
    top: 50%;
    left: 50%;
    border-radius: 50%;
    border: 3px solid #fff;
    border-top-color: transparent;
    box-sizing: border-box;
  }
  
  @-webkit-keyframes clockwise {
    from {
        transform: rotate(0deg)
    }
  
    to {
        transform: rotate(360deg)
    }
  }
  
  @-moz-keyframes clockwise {
    from {
        transform: rotate(0deg)
    }
  
    to {
        transform: rotate(360deg)
    }
  }
  
  @keyframes clockwise {
    from {
        transform: rotate(0deg)
    }
  
    to {
        transform: rotate(360deg)
    }
  }
  
  @-webkit-keyframes counter-clockwise {
    from {
        transform: rotate(0deg)
    }
  
    to {
        transform: rotate(-360deg)
    }
  }
  
  @-moz-keyframes counter-clockwise {
    from {
        transform: rotate(0deg)
    }
  
    to {
        transform: rotate(-360deg)
    }
  }
  
  @keyframes counter-clockwise {
    from {
        transform: rotate(0deg)
    }
  
    to {
        transform: rotate(-360deg)
    }
  }