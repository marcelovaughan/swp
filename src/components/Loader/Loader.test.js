import React from "react"
import { configure, mount } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import Loader from "./index";

configure({ adapter: new Adapter() });

describe("Loader Render", () => {
    it("Render the component", () => {
        const component = mount(<Loader />);
        expect(component.debug()).toMatchSnapshot();
    });
});
