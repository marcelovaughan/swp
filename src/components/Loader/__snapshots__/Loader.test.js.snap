// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Render Render the component 1`] = `
"<Loader>
  <div id=\\"spinner\\">
    <span className=\\"loader\\" />
  </div>
</Loader>"
`;
